packages/sn-filter-pane/src/components/FoldedListbox/use-delayed-hover.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from 'react';
+
+interface UseDelayedHoverProps {
+  containerRef: React.RefObject<HTMLDivElement>,
+}
+
+const useDelayedHover = ({ containerRef }: UseDelayedHoverProps) => {
+  const [isHovering, setIsHovering] = useState(false);
+  const [delayedIsHovering, setDelayedIsHovering] = useState(false);
+  const [isStillHovering, setIsStillHovering] = useState(false);
+
+  useEffect(() => {
+    const onMouseMove = (e: MouseEvent) => {
+      const elementsAtPoint = document.elementsFromPoint(e.clientX, e.clientY);
+
+      // Detect hover even when blocked by other element (blockUI in sense)
+      const mouseOver = !!elementsAtPoint.find((el) => el === containerRef.current);
+      setIsStillHovering(mouseOver);
+    };
+
+    document.addEventListener('mousemove', onMouseMove);
+    return (() => document.removeEventListener('mousemove', onMouseMove));
+  }, [containerRef.current]);
```

Comment:
do not use `containerRef.current` as a dependency
I expect it to no be needed at all
(but if you want to keep it use `containerRef` without current)